pandas-stubs/io/parsers/readers.pyi

```diff
@@ -1,618 +1,513 @@
-    parse_dates = ...
-    date_parser = ...
-    dayfirst = ...
-    keep_date_col = ...
-    na_values = ...
-    na_fvalues = ...
-    na_filter = ...
-    keep_default_na = ...
-    true_values = ...
-    false_values = ...
-    mangle_dupe_cols = ...
-    infer_datetime_format = ...
-    cache_dates = ...
-    header = ...
-    handles = ...
-    def __init__(self, kwds) -> None: ...
-    def close(self) -> None: ...
-
-class CParserWrapper(ParserBase):
-    kwds = ...
-    unnamed_cols = ...
-    names = ...
-    orig_names = ...
-    index_names = ...
-    def __init__(self, src, **kwds) -> None: ...
-    def close(self) -> None: ...
-    def set_error_bad_lines(self, status) -> None: ...
-    def read(self, nrows=...): ...
-
-def TextParser(*args, **kwds): ...
-def count_empty_vals(vals): ...
-
-class PythonParser(ParserBase):
-    data = ...
-    buf = ...
-    pos: int = ...
-    line_pos: int = ...
-    encoding = ...
-    compression = ...
-    memory_map = ...
-    skiprows = ...
-    skipfunc = ...
-    skipfooter = ...
-    delimiter = ...
-    quotechar = ...
-    escapechar = ...
-    doublequote = ...
-    skipinitialspace = ...
-    lineterminator = ...
-    quoting = ...
-    skip_blank_lines = ...
-    warn_bad_lines = ...
-    error_bad_lines = ...
-    names_passed = ...
-    has_index_names: bool = ...
-    verbose = ...
-    converters = ...
-    dtype = ...
-    thousands = ...
-    decimal = ...
-    comment = ...
-    num_original_columns = ...
-    columns = ...
-    orig_names = ...
-    index_names = ...
-    nonnum = ...
-    def __init__(self, f, **kwds): ...
-    def read(self, rows=...): ...
-    def get_chunk(self, size=...): ...
-
-class FixedWidthReader(abc.Iterator):
-    f = ...
-    buffer = ...
-    delimiter = ...
-    comment = ...
-    colspecs = ...
-    def __init__(
-        self, f, colspecs, delimiter, comment, skiprows=..., infer_nrows: int = ...
-    ) -> None: ...
-    def get_rows(self, infer_nrows, skiprows=...): ...
-    def detect_colspecs(self, infer_nrows: int = ..., skiprows=...): ...
-    def __next__(self): ...
-
-class FixedWidthFieldParser(PythonParser):
-    colspecs = ...
-    infer_nrows = ...
-    def __init__(self, f, **kwds) -> None: ...
+from collections import abc
+import csv
+from typing import (
+    Any,
+    Hashable,
+    Literal,
+    NamedTuple,
+    Sequence,
+    overload,
+)
+
+from pandas.core.frame import DataFrame
+
+from pandas._libs import lib
+from pandas._typing import (
+    CompressionOptions,
+    CSVEngine,
+    DtypeArg,
+    FilePath,
+    IndexLabel,
+    ReadCsvBuffer,
+    StorageOptions,
+)
+
+class _DeprecationConfig(NamedTuple):
+    default_value: Any
+    msg: str | None
+
+@overload
+def validate_integer(name, val: None, min_val=...) -> None: ...
+@overload
+def validate_integer(name, val: int | float, min_val=...) -> int: ...
+@overload
+def validate_integer(name, val: int | None, min_val=...) -> int | None: ...
+@overload
+def read_csv(

```

Comment:
Please make sure that the version of `read_csv()` that is currently in `parsers.pyi` is copied here as is.  I can't tell because of the file reorg.  A _lot_ of work went into getting that set of overloads correct (not to say that they couldn't be improved).  Any improvements to individual stubs should be left to a second PR.